corehq/tabs/uitab.py

```diff
@@ -126,9 +126,14 @@ def should_show(self):
         if not self.show_by_default and not self.is_active_tab:
             return False
 
-        if not self.can_access_all_locations and not self.filtered_dropdown_items:
-            # location-safe filtering makes this whole tab inaccessible
-            return False
+        if not self.can_access_all_locations:
+            if self.dropdown_items and not self.filtered_dropdown_items:
+                # location-safe filtering makes this whole tab inaccessible
+                return False
+
+            # Just a button tab, determine if it's location safe
+            if not self.dropdown_items and not url_is_location_safe(self.url):
```

Comment:
The Orgs initiative is planning to tackle this at some point soon: https://trello.com/c/mIQiaisw/25-tb-ucr-making-dashboard-location-safe